src/wp-includes/class-wp-roles.php

```diff
@@ -160,6 +160,13 @@ public function add_role( $role, $display_name, $capabilities = array() ) {
 			return;
 		}
 
+		foreach( $capabilities as $key => $value ) {
+			if(!is_bool($value)) {
+				$capabilities[$value] = true;
```

Comment:
What if `$capabilities[$value]` already exists, and what if it is `false`?

That could theoretically overwrite an intentionally blocked capability with an unintentionally allowed one.